src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/LinterRuleTestsBase.cs

```diff
@@ -17,62 +20,109 @@ public class LinterRuleTestsBase
     {
         public enum OnCompileErrors
         {
-            Fail, // Assertion fails if there are compiler errors
-            Ignore,
+            Ignore, // Ignore any compile errors
+            IncludeErrors, // Include compile errors in the list of messages to expect
+            IncludeErrorsAndWarnings, // Include compile errors and warnings in the list of messages to expect
         }
 
-        protected void AssertLinterRuleDiagnostics(string ruleCode, string bicepText, string[] expectedMessagesForCode, OnCompileErrors onCompileErrors = OnCompileErrors.Fail)
+        public enum IncludePosition
         {
-            AssertLinterRuleDiagnostics(ruleCode, bicepText, onCompileErrors, diags =>
+            None,
+            LineNumber,
+        }
+
+        private string FormatDiagnostic(IDiagnostic diagnostic, ImmutableArray<int> lineStarts, IncludePosition includePosition)
+        {
+            if (includePosition == IncludePosition.LineNumber)
             {
-                diags.Select(d => d.Message).Should().BeEquivalentTo(expectedMessagesForCode);
-            });
+                var position = TextCoordinateConverter.GetPosition(lineStarts, diagnostic.Span.Position);
+                return $"[{position.line}] {diagnostic.Message}";
+            }
+            else
+            {
+                return diagnostic.Message;
+            }
         }
 
-        protected void AssertLinterRuleDiagnostics(string ruleCode, string bicepText, int expectedDiagnosticCountForCode, OnCompileErrors onCompileErrors = OnCompileErrors.Fail)
+        static protected (string fileName, string fileContents) MainBicepFile(string bicepText)
+        {
+            return ("main.bicep", bicepText);
+        }
+
+        static protected (string fileName, string fileContents) ConfigFile(string configurationText)
+        {
+            return ("bicepconfig.json", configurationText);
+        }
+
+        static protected (string fileName, string fileContents)[] BicepFiles(string bicepText, string? configurationText)
+        {
+            List<(string fileName, string fileContents)> files = new List<(string fileName, string fileContents)>();
```

Comment:
Why does this need to be a list? We are only adding main.bicep, bicepconfig.json file and creating a new list each time.
Can't the return type be (string fileName, string fileContents)?

Reply:
removed